### PR TITLE
perf(parser): add streaming scan to ExtractExtensions for 15x speedup

### PR DESCRIPTION
**Pull request by erraggy**
## Summary

- Add `bytes.Contains` check for `"x-` pattern before JSON parsing in `ExtractExtensions`
- Avoids expensive JSON unmarshaling when no extension fields exist (common case)
- Add comprehensive tests for edge cases and benchmarks

## Benchmark Results

| Scenario | Before | After | Improvement |
|----------|--------|-------|-------------|
| **No extensions** | 2,818 ns, 65 allocs | 186 ns, 0 allocs | **15.2x faster** |
| With extensions | 2,290 ns | 2,347 ns | ~same |

## Why This Works

The pattern `"x-` (with opening quote) reliably identifies potential extension keys:
- String values like `"Use x-api-key header"` don't match (no `"x-` sequence)
- URLs like `http://x-server.com` don't match (no opening quote)
- False positives (nested keys, array elements) still return correct results

## Corpus Analysis

Most specs have few/no extensions, making this optimization highly impactful:

| Spec | Extensions |
|------|------------|
| Plaid | 0 |
| Petstore | 0 |
| Google Maps | 0 |
| Discord | 16 |
| Stripe | 2,433 |
| GitHub | 1,371 |

## Test plan

- [x] All 8,068 unit tests pass
- [x] All 15 corpus integration tests pass
- [x] New edge case tests cover: nested keys, array elements, false positives, minimum extension names
- [x] Benchmarks added to track performance over time

## Design Doc

See `docs/plans/2026-01-30-extract-extensions-optimization-design.md` for analysis of alternative approaches (Option 2: single-parse refactor) and why they were deferred.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Documentation**
  * Added a design doc describing the optimization approach, phases, and benchmark analysis.

* **Performance**
  * Reduced JSON processing CPU cost with a streaming pre-check that short-circuits work when no extensions are present, yielding measurable speedups.

* **Tests**
  * Added comprehensive tests and benchmarks covering edge cases, integration round-trips, and performance scenarios.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: docs/plans/2026-01-30-extract-extensions-optimization-design.md [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+- All 15 corpus integration tests pass
+- New edge case tests cover: nested keys, array elements, false positives, minimum extension names
+
+### Benchmarks
+
+```
+BenchmarkExtractExtensions_NoExtensions     186.0 ns/op    0 B/op    0 allocs/op
+BenchmarkExtractExtensions_WithExtensions  2347 ns/op   2512 B/op   57 allocs/op
+BenchmarkExtractExtensions_FalsePositive    108 ns/op      0 B/op    0 allocs/op
+```
```
[the remark this answers is not in the document]
Comment (erraggy) [reply docs/plans/2026-01-30-extract-extensions-optimization-design.md]: this plan document is only needed to plan, and not to document long-term.
